apiserver/facades/client/client/instanceconfig.go

```diff
@@ -77,7 +77,7 @@ func InstanceConfig(st *state.State, machineId, nonce, dataDir string) (*instanc
 	caCert, _ := controllerConfig.CACert()
 
 	// Get the API connection info; attempt all API addresses.
-	apiHostPorts, err := st.APIHostPorts()
+	apiHostPorts, err := st.APIHostPortsForAgents()
```

Comment:
This, I believe, is indeed valid.

BTW, can you audit to see if Client.InstanceConfig is actually being used in any client code?
I'm trying to remember the case where the *client* is responsible for starting an instance once we have the controller running. (It may be something like manual provisioning, etc.)
Though if we aren't using it now, ideally we would flag this method as deprecated, and remove it from new versions of the Client facade, so we don't have to maintain it in the future.

Reply:
It is only used by _environs/manual/sshprovisioner/provisioner_test.go_.
I will see to your suggestion.